R/plotLoadings.R

```diff
@@ -0,0 +1,416 @@
+#'   (default: \code{rownames})
+#'   
+#' @param ... additional arguments for plotting. See 
+#'   \code{\link{mia-plot-args}} for more details i.e. call \code{help("mia-plot-args")}     
+#' 
+#' @details
+#' 
+#' Inspired by the \code{\link[diffTop:plotASVcircular]{plotASVcircular}} method using phyloseq 
+#' and has been converted to use TreeSummarizedExperiment/SingleCellExperiment objects.
+#' 
+#' TreeSummarizedExperiment/SingleCellExperiment objects are expected to have content in reducedDim slot.
+#' It is impossible to plot tree if only the matrix is given. Number of features must be reduced
+#' before calling function or it will not be understandable. For example
+#' agglomerating data by rank or prevalence (see examples).
+#' 
+#' @return 
+#' A \code{ggplot2} object. A circular plot annotated with TreeSummarizedExperiment object.
+#'
+#' @name plotLoadings
+#' @export
+#' 
+#' @author
+#' Ely Seraidarian
+#' Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' 
+#' # Plotting feature loadings with tree
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- transformAssay(tse, method = "clr", pseudocount = 1)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", update.tree = TRUE)
+#' tse <- runPCA(tse, ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse)
+#' 
+#' # Plotting without tree as a heatmap
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix)
+#' 
+#' # Plotting without tree as a barplot
+#' plotLoadings(loadings_matrix, layout = "barplot")
+#' 
+#' # Plotting more features
+#' plotLoadings(loadings_matrix, n = 12)
+#'
+#' # Plotting with less components
+#' tse <- runPCA(tse, ncomponents = 4, assay.type = "clr")
+#' loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+#' plotLoadings(loadings_matrix, ncomponents = 4)
+#' 
+#' # Plotting if loadings matrix name has been changed
+#' tse <- runPCA(tse, name = "myPCAmatrix", ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, dimred = "myPCAmatrix")
+#' 
+#' # Plotting tree with taxonomic rank classification
+#' tse <- runPCA(tse, ncomponents = 5, assay.type = "clr")
+#' plotLoadings(tse, class = "Phylum")
+#' 
+#' # Plotting after performing LDA method
+#' tse <- addLDA(tse)
+#' plotLoadings(tse, dimred = "LDA", ncomponents = 2)
+NULL
+
+#' @rdname plotLoadings
+setGeneric("plotLoadings", signature = c("x"),
+    function(x, ...) 
+        standardGeneric("plotLoadings"))
+
+
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "TreeSummarizedExperiment"),
+    function(x,
+            dimred = "PCA",
+            layout = "tree",
+            n = 10,
+            ncomponents = 5,
+            tree.name = "phylo",
+            class = rownames(x),
+            ...) {
+      
+        if (length(reducedDimNames(x)) == 1) {
+            dimred <- reducedDimNames(x)
+        }
```

Comment:
Maybe I could still keep it so that user do not have to specify dimred when only one reduction method has been performed (removed default value though)